### PR TITLE
implementation

### PR DESCRIPTION
**Pull request by TomasTomecek**

### CODE REVIEW

File: docker.yaml [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+        # Build dependencies of this module, optional
+        # Keys are module names, values are the stream names
+        # These modules define the buildroot for this module
+        buildrequires:
+            brt: something
+            golang: something-else
+        # Run-time dependencies of this module, optional
+        # Keys are module names, values are their stream names
+        requires:
+            i-dont: know
```
Comment (nphilipp) [inline docker.yaml]: Heh. Probably leave "requires" empty?

File: docker.yaml [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+        # Content license, optional
+        # A list of licenses used by the packages in the module.
+        # This should be populated by build tools.
+        content:
+            - ASL 2.0
+    # Extensible metadata block
+    # A dictionary of user-defined keys and values.
+    # Optional.  Defaults to an empty dictionary.
+    xmd:
+      some_key: some_data
```
Comment (nphilipp) [inline docker.yaml]: I'd leave the xmd block empty, as well as everything else we don't know that we need it.

File: docker.yaml [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+            rpms:
+                - docker
+        vim:
+            description: Profile for vim users which installs Dockerfile syntax support for vim.
+            rpms:
+                - docker-vim
+        zsh:
+            description: Profile for zsh users which installs completion.
+            rpms:
+                - docker-zsh-completion
```
Comment (nphilipp) [inline docker.yaml]: Ahh, there's your use case for multiple install profiles side-by-side. :)

File: docker.yaml [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+        rpms:
+            - docker
+            - docker-common
+            - docker-rhel-push-plugin
+            - docker-vim
+            - docker-zsh-completion
+
+    # Module component filters
+    filter:
+        # RPM names not to be included in the module.
```
Comment (nphilipp) [inline docker.yaml]: If I understand this correctly, YAML needs a "~" for "this is empty". Maybe it's just by chance that the Python yaml module doesn't trip over this?

Comment (TomasTomecek) [inline docker.yaml]: Highly possible. When I set `?` as a value of dict, it went completely mayhem.

File: docker.yaml (hunk 3 above)
Comment (TomasTomecek) [inline docker.yaml]: Yep, I like them!!

File: docker.yaml [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+                # Use this lookaside cache if it's different from the
+                # build system configuration.
+                # Optional.
+                # cache: https://example.com/cache
+                # Use this specific commit has, branch name or tag for
+                # the build.  If ref is a branch name, the branch HEAD
+                # will be used.  If no ref is given, the master branch
+                # is assumed.
+                # Optional.
+                ref: master
```
Comment (TomasTomecek) [inline docker.yaml]: So, I understand components as SRPM and api as binary RPMs -- all the packages from API are built from this SRPM.

**Comment by TomasTomecek** [comment]
Thanks @nphilipp! I addressed most, expect for the first one: the `components` section.

Honestly, I'm waiting now for @FrostyX to enable module builds in COPR so I can try this out.

File: docker.yaml (hunk 5 above)
Comment (nphilipp) [inline docker.yaml]: Ahh right.

File: docker.yaml (hunk 4 above)
Comment (nphilipp) [inline docker.yaml]: :smile:

**Comment by nphilipp** [comment]
Sorry for the long wait. Looks good to me! I'd merge this as a single commit, if it's alright with you.

**Comment by TomasTomecek** [comment]
@nphilipp thanks; feel free to go ahead

**Comment by nphilipp** [comment]
Merged this:

commit 45e04a720010b590e2a3255da6cf65841e1a3266
Author:     Tomas Tomecek <ttomecek@redhat.com>
AuthorDate: Wed Mar 1 15:53:02 2017 +0100
Commit:     Nils Philippsen <nils@redhat.com>
CommitDate: Fri Mar 10 12:35:57 2017 +0100

    add modulemd
    
    Signed-off-by: Tomas Tomecek <ttomecek@redhat.com>
    Signed-off-by: Nils Philippsen <nils@redhat.com>

**Comment by TomasTomecek** [comment]
Thank you! Let's close then.

**Comment by nphilipp** [comment]
Ah yeah, missed that somehow :).

**Pull request closed without merging**